client-core/src/handler/default_block_handler.rs

```diff
@@ -93,13 +91,7 @@ where
                 )?;
             }
         }
-
-        self.global_state_service.set_global_state(
```

Comment:
I don't think this should be removed from here. (Reasons are mentioned in a different comment)